src/stores/modalContext.tsx

```diff
@@ -0,0 +1,235 @@
+import React, { createContext, useContext, useRef, useState } from 'react'
+import { ModalRef } from '@trussworks/react-uswds'
+import { useAnalytics } from 'stores/analyticsContext'
+import { useAddBookmarkMutation } from 'operations/portal/mutations/addBookmark.g'
+import { useRemoveCollectionMutation } from 'operations/portal/mutations/removeCollection.g'
+import { useRemoveWidgetMutation } from 'operations/portal/mutations/removeWidget.g'
+import { useRemoveBookmarkMutation } from 'operations/portal/mutations/removeBookmark.g'
+import { useEditBookmarkMutation } from 'operations/portal/mutations/editBookmark.g'
+import { Widget, Bookmark as BookmarkType } from 'types/index'
+
+// Try to clean this up
+export type ModalContextType = {
+  modalId: string
+  updateModalId: (modalId: string) => void
+  modalRef: React.RefObject<ModalRef> | null | any
```

Comment:
I had to add the `any` type here because there is a mismatch between this type, and the mocked ref in `testHelpers.tsx` on line 11. I couldn't figure out how to update the type and the mismatch caused several tests to fail, so I thought this could be an interim solution.